Procurement/Controls/FragmentStashViewModel.cs

```diff
@@ -0,0 +1,62 @@
+
+        public ItemDisplayViewModel Grief => GetItemCalled("Mortal Grief");
+        public ItemDisplayViewModel Rage => GetItemCalled("Mortal Rage");
+        public ItemDisplayViewModel Ignorance => GetItemCalled("Mortal Ignorance");
+        public ItemDisplayViewModel Hope => GetItemCalled("Mortal Hope");
+
+        public ItemDisplayViewModel Volkuur => GetItemCalled("Volkuur's Key");
+        public ItemDisplayViewModel Eber => GetItemCalled("Eber's Key");
+        public ItemDisplayViewModel Yriel => GetItemCalled("Yriel's Key");
+        public ItemDisplayViewModel Inya => GetItemCalled("Inya's Key");
+
+        public ItemDisplayViewModel Hydra => GetItemCalled("Fragment of the Hydra");
+        public ItemDisplayViewModel Phoenix => GetItemCalled("Fragment of the Phoenix");
+        public ItemDisplayViewModel Minotaur => GetItemCalled("Fragment of the Minotaur");
+        public ItemDisplayViewModel Chimera => GetItemCalled("Fragment of the Chimera");
+
+        public ItemDisplayViewModel DivineVessel => GetItemCalled("Divine Vessel");
+
+        public ItemDisplayViewModel ChayulaSplinter => new ItemDisplayViewModel(_stash.OfType<Splinter>().FirstOrDefault(x => x.Type == BreachType.Chayula));
+        public ItemDisplayViewModel EshSplinter => new ItemDisplayViewModel(_stash.OfType<Splinter>().FirstOrDefault(x => x.Type == BreachType.Esh));
+        public ItemDisplayViewModel TulSplinter => new ItemDisplayViewModel(_stash.OfType<Splinter>().FirstOrDefault(x => x.Type == BreachType.Tul));
+        public ItemDisplayViewModel UulNetolSplinter => new ItemDisplayViewModel(_stash.OfType<Splinter>().FirstOrDefault(x => x.Type == BreachType.UulNetol));
+        public ItemDisplayViewModel XophSplinter => new ItemDisplayViewModel(_stash.OfType<Splinter>().FirstOrDefault(x => x.Type == BreachType.Xoph));
+
+        public ItemDisplayViewModel ChayulaBlessing => new ItemDisplayViewModel(_stash.OfType<BreachBlessing>().FirstOrDefault(x => x.Type == BreachType.Chayula));
+        public ItemDisplayViewModel EshBlessing => new ItemDisplayViewModel(_stash.OfType<BreachBlessing>().FirstOrDefault(x => x.Type == BreachType.Esh));
+        public ItemDisplayViewModel TulBlessing => new ItemDisplayViewModel(_stash.OfType<BreachBlessing>().FirstOrDefault(x => x.Type == BreachType.Tul));
+        public ItemDisplayViewModel UulNetolBlessing => new ItemDisplayViewModel(_stash.OfType<BreachBlessing>().FirstOrDefault(x => x.Type == BreachType.UulNetol));
+        public ItemDisplayViewModel XophBlessing => new ItemDisplayViewModel(_stash.OfType<BreachBlessing>().FirstOrDefault(x => x.Type == BreachType.Xoph));
+
+        public ItemDisplayViewModel Offering => new ItemDisplayViewModel(_stash.OfType<Offering>().FirstOrDefault());
+
+
+        private ItemDisplayViewModel GetItemCalled(string name)
+        {
+            var item = _stash.FirstOrDefault(x => x.DescriptiveName.Equals(name, StringComparison.CurrentCultureIgnoreCase));
```

Comment:
Don't use the `DescriptiveName` property here.  For one, it's fairly expensive to compute that property, since all the components are regenerated every time.  Also, there's no guarantee the value returned will equal the type line (that is, the value you're comparing against), since there's no stated guarantee on what the `DescriptiveName` is.  It's meant to be a string that can be presented in the UI.